### PR TITLE
Drop obviously wrong boolean assertions usage

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: core/src/test/java/org/apache/james/core/MailAddressTest.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public void testHashCode() throws AddressException {
     public void testMailAddressString() throws AddressException {
 
         MailAddress a = new MailAddress(GOOD_ADDRESS);
-        assertThat(GOOD_ADDRESS.equals(a.toString())).isTrue();
+        assertThat(GOOD_ADDRESS).isEqualTo(a.toString());
```
Comment (rouazana) [inline core/src/test/java/org/apache/james/core/MailAddressTest.java]: should be reversed, no?

File: core/src/test/java/org/apache/james/core/MailAddressTest.java [hunk 2]
```diff
@@ -223,8 +223,7 @@ public void testEqualsObject() throws AddressException {
         MailAddress a = new MailAddress(GOOD_ADDRESS);
         MailAddress b = new MailAddress(GOOD_ADDRESS);
 
-        assertThat(a.equals(b)).describedAs(a.toString() + " != " + b.toString()).isTrue();
-        assertThat(a.equals(null)).describedAs(a.toString() + " != " + null).isFalse();
+        assertThat(a).isNotNull().isEqualTo(b);
```
Comment (rouazana) [inline core/src/test/java/org/apache/james/core/MailAddressTest.java]: add some cr/lf?

File: mailbox/hbase/src/test/java/org/apache/james/mailbox/hbase/mail/HBaseMailboxMapperTest.java [hunk 3]
```diff
@@ -314,7 +314,7 @@ private void testChunkStream() throws IOException {
                 bout.write(b);
             }
             String s = bout.toString();
-            assertThat(original.equals(s)).isTrue();
+            assertThat(original).isEqualTo(s);
```
Comment (rouazana) [inline mailbox/hbase/src/test/java/org/apache/james/mailbox/hbase/mail/HBaseMailboxMapperTest.java]: inversed?

**Comment by mbaechler** [comment]
> I don't see any dropping of assert(true).isFalse, is it expected?

I looked at some but it's always tests that should use assertThatThrownBy and it's way harder to handle.
In a next step maybe.

**Comment by mbaechler** [comment]
```
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] [ERROR] connectingToAClusterWithAFailedRabbit{DockerRabbitMQCluster}  Time elapsed: 40.114 s  <<< ERROR!
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] java.io.IOException
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] 	at org.apache.james.queue.rabbitmq.RabbitMQClusterTest$ClusterNodesFailure.setup(RabbitMQClusterTest.java:165)
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] Caused by: com.rabbitmq.client.ShutdownSignalException: connection error
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] 	at org.apache.james.queue.rabbitmq.RabbitMQClusterTest$ClusterNodesFailure.setup(RabbitMQClusterTest.java:165)
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] Caused by: java.net.SocketException: Connection reset
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] 
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] [ERROR] rabbitMQManagerShouldReturnThreeNodesWhenAskingForStatus{DockerRabbitMQCluster}  Time elapsed: 41.372 s  <<< ERROR!
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] java.io.IOException
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] 	at org.apache.james.queue.rabbitmq.RabbitMQClusterTest$ClusterSharing.setup(RabbitMQClusterTest.java:82)
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] Caused by: com.rabbitmq.client.ShutdownSignalException: connection error
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] 	at org.apache.james.queue.rabbitmq.RabbitMQClusterTest$ClusterSharing.setup(RabbitMQClusterTest.java:82)
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] Caused by: java.net.SocketException: Connection reset
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] 
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] [INFO] 
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] [INFO] Results:
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] [INFO] 
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] [ERROR] Errors: 
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] [ERROR]   RabbitMQClusterTest$ClusterNodesFailure.setup:165 » IO
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] [ERROR]   RabbitMQClusterTest$ClusterSharing.setup:82 » IO
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] [INFO] 
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] [ERROR] Tests run: 30, Failures: 0, Errors: 2, Skipped: 1
[b5d5d3c29229474a0ba8d539fdc01774a0284bee] [INFO] 
```


test this please

**Pull request merged**